arangod/MMFiles/MMFilesCollection.cpp

```diff
@@ -181,7 +181,11 @@ Result persistLocalDocumentIdIterator(MMFilesMarker const* marker, void* data,
         localDocumentId = LocalDocumentId(
             encoding::readNumber<LocalDocumentId::BaseType>(ptr, sizeof(LocalDocumentId::BaseType)));
       } else {
-        localDocumentId = LocalDocumentId::create();
+        VPackSlice keySlice;
+        TRI_voc_rid_t revisionId;
+        transaction::helpers::extractKeyAndRevFromDocument(slice, keySlice, revisionId);
+
+        localDocumentId = LocalDocumentId::create(revisionId);
```

Comment:
I think we can basically leave mmfiles as it is and just not use the replication protocol, wdyt ?

Reply:
I would be perfectly happy with that, to be honest. @jsteemann?